IPAdapterPlus.py

```diff
@@ -427,24 +471,37 @@ def apply_ipadapter(self, ipadapter, model, weight, clip_vision=None, image=None
             clip_embed = embeds[0].cpu()
             clip_embed_zeroed = embeds[1].cpu()
         else:
-            if image.shape[1] != image.shape[2]:
-                print("\033[33mINFO: the IPAdapter reference image is not a square, CLIPImageProcessor will resize and crop it at the center. If the main focus of the picture is not in the middle the result might not be what you are expecting.\033[0m")
+            if self.is_face_id:
+                global IPADAPTER_INSIGHT_FACE_APP
+                if IPADAPTER_INSIGHT_FACE_APP is None:
+                    app = FaceAnalysis(name="buffalo_l", providers=['ROCMExecutionProvider', 'CPUExecutionProvider'])
```

Comment:
This needs insightface installed to work which uses onnxruntime. On my computer onnxruntime is extremely slow to startup for some reason and it can only find CPU provider (I'm on linux with rocm)

So I just cached it like this, but probably there is a better way to do this. Maybe even a separate node.

Reply:
onnx will only work at the moment with cuda118. Have you verified you aren't on the newest version? I had to manage this while installing insight face earlier.